abci/example/kvstore/kvstore.go

```diff
@@ -166,34 +167,64 @@ func (app *Application) InitChain(_ context.Context, req *types.InitChainRequest
 // - `=` is not the first or last byte.
 // - if key is `val` that the validator update transaction is also valid.
 func (app *Application) CheckTx(_ context.Context, req *types.CheckTxRequest) (*types.CheckTxResponse, error) {
+	code := CodeTypeOK
 	// If it is a validator update transaction, check that it is correctly formatted
 	if isValidatorTx(req.Tx) {
 		if _, _, _, err := parseValidatorTx(req.Tx); err != nil {
-			if app.useLanes {
-				return &types.CheckTxResponse{Code: CodeTypeInvalidTxFormat, Lane: app.lanes["val"]}, nil
-			}
-			return &types.CheckTxResponse{Code: CodeTypeInvalidTxFormat}, nil
+			code = CodeTypeInvalidTxFormat
 		}
 	} else if !isValidTx(req.Tx) {
-		return &types.CheckTxResponse{Code: CodeTypeInvalidTxFormat}, nil
+		code = CodeTypeInvalidTxFormat
```

Comment:
If it is invalid should we not return here immediately 
```suggestion
		return &types.CheckTxResponse{Code: CodeTypeInvalidTxFormat, Lane: uint32(0) }, nil
```

Reply:
Yes, probably that's better. I will change it.